### PR TITLE
Task Editor

### PR DESCRIPTION
**Pull request by jlmjkfd**
Most functions of the Task Editor except the project selector.
Move the Task Editor components into the same folder.
Please change the import path if the TaskEditor has been used before.

### CODE REVIEW

File: components/application/widgets/taskEditor/TaskEditor.js [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+  }
+  if (submitCallBack === null) {
+    submitCallBack = () => {};
+  }
+
+  const taskNameInputRef = useRef(null);
+
+  /** record the editing task */
+  let newTaskData = useRef({
+    id: taskData.id === null ? "" : taskData.id,
```
Comment (ryanwooa) [inline components/application/widgets/taskEditor/TaskEditor.js:64]: When I Click addTask button, it show this Error
image

Comment (jlmjkfd) [reply components/application/widgets/taskEditor/TaskEditor.js]: fixed

**Review by zli213 — COMMENTED** [review]
There is no apparent error in the provided code.

File: styles/scss/components/application/widgets/priorityPicker.module.scss [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    z-index: 1000;
+}
+
+.priority_picker {
+    position: absolute;
+    display: flex;
+    flex-direction: column;
+    background-color: #fff;
+    border-radius: 0.8rem;
+    box-shadow: 0px 1px 2rem 1px #c4c4c4;
```
[the remark this answers is not in the document]
Comment (jlmjkfd) [reply styles/scss/components/application/widgets/priorityPicker.module.scss:18]: Which common style do you mean?

[the remark this answers is not in the document]
Comment (zli213) [reply styles/scss/components/application/widgets/priorityPicker.module.scss:18]: Colours and border-radius. Also, consider applying a unified style for other similar style files, if possible.

[the remark this answers is not in the document]
Comment (jlmjkfd) [reply styles/scss/components/application/widgets/priorityPicker.module.scss:18]: done

**Review by ryanwooa — APPROVED** [review]
No obvious problem

**Review by zli213 — APPROVED** [review]
No obvious problems.

**Review by kesesek — COMMENTED** [review]
Please solve the conflict. No other problems.

**Pull request merged**